testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/metrics/rbac/MicroProfileMetricsRbacTestCase.java

```diff
@@ -0,0 +1,170 @@
+        client.close();
+        setRbacStatus("simple");
+        setEndpointSecurityStatus(false);
+
+        ServerReload.reloadIfRequired(mgmtClient);
+    }
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class,
+                MicroProfileMetricsRbacTestCase.class.getSimpleName() + ".war")
+                .addClasses(TestApplication.class)
+                ;
+        return war;
+    }
+
+    @Test
+    @InSequence(1)
+    public void testAuthenticatedMetricsRequestWithInvalidRole() throws Exception {
+        final CloseableHttpResponse response = makeRequest(metricsUrl, getHttpClientContext());
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        final String entity = EntityUtils.toString(response.getEntity());
+        assertFalse(entity.contains("wildfly"));
+    }
+
+    @Test
+    @InSequence(2)
+    public void testAuthenticatedMetricsRequestWithValidRole() throws Exception {
+        mapUserRole();
+
+        final CloseableHttpResponse response = makeRequest(metricsUrl, getHttpClientContext());
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        final String entity = EntityUtils.toString(response.getEntity());
+        assertTrue(entity.contains("wildfly"));
+    }
+
+    private HttpClientContext getHttpClientContext() {
+        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials("testSuite", "testSuitePassword"));
+        HttpClientContext hcContext = HttpClientContext.create();
+        hcContext.setCredentialsProvider(credentialsProvider);
+        return hcContext;
+    }
+
+    private void setRbacStatus(String provider) throws IOException {
+        executeOperation(Util.getWriteAttributeOperation(
+                PathAddress.parseCLIStyleAddress("/core-service=management/access=authorization"),
+                "provider", provider)
+        );
+    }
+
+    private void setEndpointSecurityStatus(boolean status) throws IOException {
+        executeOperation(Util.getWriteAttributeOperation(
+                PathAddress.parseCLIStyleAddress("/subsystem=microprofile-metrics-smallrye"),
+                "security-enabled", status)
+        );
+    }
+
+    private void mapUserRole() throws IOException {
+        executeOperation(Util.createAddOperation(
+                PathAddress.parseCLIStyleAddress("/core-service=management/access=authorization/role-mapping=Monitor"))
+        );
+        final ModelNode addOperation = Util.createAddOperation(
+                PathAddress.parseCLIStyleAddress("/core-service=management/access=authorization/role-mapping=Monitor/include=monitor")
+        );
+        addOperation.get("alias").set("testSuite");
+        addOperation.get("name").set("testSuite");
+        addOperation.get("type").set("USER");
+        executeOperation(addOperation);
```

Comment:
Again not critical here as since it's only two operations it's not too slow, but you can do composite operations so only one request is made.

```java
final CompositeOperationBuilder builder = CompositeOperationBuilder.create();
builder.addStep(Operations.createAddOperation(
        Operations.createAddress("core-service", "management", "access", "authorization", "role-mapping", "Monitor")));

final ModelNode addOperation = Operations.createAddOperation(
        Operations.createAddress("core-service", "management", "access=authorization", "role-mapping", "Monitor", "include", "monitor")
);
addOperation.get("alias").set("testSuite");
addOperation.get("name").set("testSuite");
addOperation.get("type").set("USER");
builder.addStep(addOperation);
executeOperation(builder.build());
```

Reply:
I like this change, but it would require I duplicate executeOperation() to take and Operation, so I'll probably leave it as is. Leaves the code simpler at the cost of a few ms. :)

Reply:
FWIW what I usually do is:
```
private void executeOperation(final ModelNode op) {
    executeOperation(Operation.Factory.create(op)):
}
```
Then just have the one that accepts `Operation` do the logic.

Reply:
Pfft. I like that. Incoming. :)